core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -320,7 +320,28 @@ private[spark] object Utils extends Logging {
    * newly created, and is not marked for automatic deletion.
    */
   def createDirectory(root: String, namePrefix: String = "spark"): File = {
-    JavaUtils.createDirectory(root, namePrefix)
+    var attempts = 0
```

Comment:
If we believe that `createDirectory` has no problem, we can only revert `createTempDir` part

Reply:
I think this part don't needs to be changed